printing/proof_diffs.ml

```diff
@@ -660,3 +663,22 @@ let make_goal_map op np =
   let ng_to_og = make_goal_map_i op np in
   (*db_goal_map op np ng_to_og;*)
   ng_to_og
+
+let diff_proofs ~diff_opt ?old proof =
+  let pp_proof p =
+    let sigma, env = Proof.get_proof_context p in
+    let pprf = Proof.partial_proof p in
+    Pp.prlist_with_sep Pp.fnl (pr_econstr_env env sigma) pprf in
```

Comment:
This can be floated to the top-level.

Reply:
Where would you move it?  It's called from both `Coqloop` and `Idetop`.  I don't think there should be a caller/callee relationship between those.  And it makes sense to keep the diffs code routines together when possible.

Reply:
I mean the `pp_proof` subfunction, it is creating a closure needlessly, tho the optimizer compiler will float it.

Reply:
I see what you mean.  For modularity, I like to make such helper routines as inner routines if they're only used by one routine.  That seems to be a common pattern in Coq.  The cost of a closure is pretty small, and this UI code is performance constrained.  Does it really matter?

Reply:
As you prefer, if you like it this way it's fine.